FormalConjectures/ErdosProblems/123.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 123
+
+*References:*
+- [erdosproblems.com/123](https://www.erdosproblems.com/123)
+- [ErLe96] Erdős, P. and Lewin, Mordechai, _$d$-complete sequences of integers_. Math. Comp. (1996), 837-840.
+- [Er92b] Erdős, Paul, _Some of my favourite problems in various branches of combinatorics_. Matematiche (Catania) (1992), 231-240.
+-/
+
+open Filter
+
+namespace Erdos123
+
+/--
+A set `A` of natural numbers is **d-complete** if every sufficiently large integer
+is the sum of distinct elements of `A` such that no element divides another.
+
+Reference: [ErLe96] Erdős, P. and Lewin, M., _$d$-complete sequences of integers_. Math. Comp. (1996).
+-/
+def IsDComplete (A : Set ℕ) : Prop :=
+  ∀ᶠ n in atTop, ∃ s : Finset ℕ,
+    (s : Set ℕ) ⊆ A ∧                    -- The summands come from A
+    IsAntichain (· ∣ ·) (s : Set ℕ) ∧   -- No summand divides another
+    s.sum id = n                         -- They sum to n
+
+/--
+The set of all natural numbers of the form $p^a q^b$ where $a, b ≥ 0$.
+This represents numbers whose prime factors are restricted to those of $p$ and $q$.
+-/
+def PowersOfTwo (p q : ℕ) : Set ℕ :=
+  {n : ℕ | ∃ a b : ℕ, n = p^a * q^b}
+
+/--
+The set of all natural numbers of the form $p^a q^b r^c$ where $a, b, c ≥ 0$.
+This represents numbers whose prime factors are restricted to those of $p$, $q$, and $r$.
```

Comment:
Note that the set you've defined isn't always the same as this set, since you need to account for multiplicities. For example, taking `(p, q, r) = (2^2, 3^2, 5^2)` should give you a counterexample.